### PR TITLE
test: directory targets and shared cache

### PR DESCRIPTION
**Pull request by emillon**
See #8067.

### CODE REVIEW

**Comment by Alizter** [comment]
Would it be useful to have a regular file that also prints something on creation so that we can compare the behaviour for files side by side with the behaviour for directories?

**Comment by rgrinberg** [comment]
There is some overlap with what `main.t` is testing. Could you move (or delete if it's redundant) what's being tested there into your test?

**Comment by emillon** [comment]
@rgrinberg done.
@Alizter presumably this is already being tested elsewhere; this test is specific to the interaction between directory targets and the shared cache.

**Review by rgrinberg — APPROVED** [review]

File: test/blackbox-tests/test-cases/directory-targets/cache.t [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  $ dune_cmd exists _build/default/out/b
+  true
+  $ dune cache size
+  0B
+
+Now we create another project with the same contents.
+
+  $ cd ..
+  $ mkdir b
+  $ cp -t b a/dune-project a/dune a/create.sh
```
Comment (rgrinberg) [inline test/blackbox-tests/test-cases/directory-targets/cache.t:46]: I see `-t` doesn't exist on macos.

Btw, why not just `mv`? I see you're not using the old project anyway.

Comment (emillon) [reply test/blackbox-tests/test-cases/directory-targets/cache.t]: Ah I thought it was a posix thing; I switched to the other form.

**Pull request merged**